File: basic_library/src/main/java/com/xiaoxin/library/base/BaseLazyFragment.java
package com.xiaoxin.library.base;

import android.util.Log;

public abstract class BaseLazyFragment extends BaseFragment {

    private static final String TAG = "BaseLazyFragment";

    private boolean isDataLoaded;

    public abstract void doLazyBusiness();

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        Log.d(TAG, "setUserVisibleHint: " + isVisibleToUser);
        super.setUserVisibleHint(isVisibleToUser);
        if (isVisibleToUser && mContentView != null && !isDataLoaded) {
            doLazyBusiness();
            isDataLoaded = true;
        }
    }

    @Override
    public void doBusiness() {
        if (getUserVisibleHint() && !isDataLoaded) {
            doLazyBusiness();
            isDataLoaded = true;
        }
    }
}

File: basic_library/src/main/java/com/xiaoxin/library/base/BaseLazyFragment.kt
package com.xiaoxin.library.base

import android.util.Log

abstract class BaseLazyFragment : BaseFragment() {

    private var isDataLoaded: Boolean = false

    abstract fun doLazyBusiness()

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        Log.d(TAG, "setUserVisibleHint: $isVisibleToUser")
        super.setUserVisibleHint(isVisibleToUser)
        if (isVisibleToUser && mContentView != null && !isDataLoaded) {
            doLazyBusiness()
            isDataLoaded = true
        }
    }

    override fun doBusiness() {
        if (userVisibleHint && !isDataLoaded) {
            doLazyBusiness()
            isDataLoaded = true
        }
    }

    companion object {

        private val TAG = "BaseLazyFragment"
    }
}
